common-content/en/module/tools/shell-pipelines/index.md

```diff
@@ -5,11 +5,38 @@ time = 30
 facilitation = false
 emoji= "💻"
 [objectives]
-    1="Count the occurences of different lines within a file using sort and uniq"
-    2="List the files in a directory which contain an upper-case letter in their name with ls and grep"
-    3="Count the number of files in a directory which contain an upper-case letter in their name with ls, grep, and wc"
-    4="Replace all occurrences of one character with another using tr"
-    5="TODO: Pipe the output of jq to something interesting"
+    1="Describe what a shell pipeline is."
+    2="Explain why we use shell pipelines."
+    3="Explain the difference between stdout and stderr."
+    4="Explain what gets passed between two programs when they're combined with a `|`."
 +++
 
-### Shell tools
\ No newline at end of file
+Read the learning objectives listed on this page: Bear in mind what you’re trying to achieve while reading this text. If a topic isn’t making much sense, and isn’t in the objectives, you can probably skip over it. If a topic is listed in the objectives, you should keep studying it until you are confident you’ve met the objective.
+
+{{<note type="Reading">}}
+Read [Thinking in Pipelines from Effective Shell](https://effective-shell.com/part-2-core-skills/thinking-in-pipelines/).
+{{</note>}}
+
+### Key take-aways:
+* Most programs take some input, and produce some output.
+* Instead of writing one program to do exactly what we want, we can often combine existing programs.
+* Programs read from stdin, and write to stdout and stderr.
+  * They write their main output to stdout.
+  * They write error messages, progress messages, and other information that isn't their main output to stderr.
+* We can pass information between programs using a pipe: `|`
+* We can write the output of programs to a file using `>`, or append to a file using `>>`.
+
+{{<multiple-choice
+   question="If /doesnotexist doesn't exist, what will be output to stdout and stderr by the command `ls /doesnotexist`"
+   answers="stdout: an error message. stderr: nothing. | stdout: /doesnotexist. stderr: an error message. | stdout: nothing. stderr: an error message."
+   feedback="Not quite - what are stdout and stderr for? | Not quite - ls only lists files that exist. | Right - ls doesn't have any files to list as output, but does have an error to display."
+   correct="2" >}}
+
+{{<multiple-choice
```

Comment:
definitely would like a diagram here showing the product of an operation being passed to the next operation

I would also say can you relate to prior learning? 

pipe | is like | chained functions | array.map().filter()